src/InstrumentationEngine/ProfilerManager.cpp

```diff
@@ -3706,12 +3706,14 @@ HRESULT CProfilerManager::AddMethodInfoToMap(_In_ FunctionID functionId, _In_ CM
         return E_FAIL;
     }
 
+    CCriticalSectionHolder holder(&m_cs);
     m_methodInfos.insert({ functionId, pMethodInfo });
     return S_OK;
 }
 
 HRESULT CProfilerManager::RemoveMethodInfoFromMap(_In_ FunctionID functionId)
 {
+    CCriticalSectionHolder holder(&m_cs);
```

Comment:
>m_cs [](start = 35, length = 4)

I'm a little concerned about deadlocks because we take a critical section on m_csForJit before calling this function

Reply:
Looks like it should be OK, since nothing that calls the jit functions takes this critical section. However, these calls may be blocked by unrelated functions in the profiler manager. Maybe it is better to create a critical section for guarding this collection specifically?

---
In reply to: [603663599](https://github.com/microsoft/CLRInstrumentationEngine/pull/385#discussion_r603663599) [](ancestors = 603663599)